.changeset/silly-mirrors-marry.md

```diff
@@ -0,0 +1,6 @@
+---
+"@latticexyz/store": minor
+"@latticexyz/world": patch
+---
+
+Added an `abstract` `StoreWrite` contract, and renamed `StoreData` to `Store`.
```

Comment:
```suggestion
Added an `abstract` `StoreKernel` contract, which includes all Store interfaces except for registration, and implements write methods, `protocolVersion` and initializes `StoreCore`. `Store` extends `StoreKernel` with the `IStoreRegistration` interface. `StoreData` is removed as a separate interface/contract. `World` now extends `StoreKernel` (since the registration methods are added via the `InitModule`).
```